### PR TITLE
Increase JAXB extension test coverage

### PR DESCRIPTION
**Pull request by jamesnetherton**
Note: `impsort-maven-plugin` upgraded as the older version does not seem to deal with `jaxb.index` properly.

### CODE REVIEW

**Comment by aldettinger** [comment]
There are usages of `java.lang.reflect.Method` in camel `JaxbDataFormat.java`, `JaxbHelper.java` and `FallBackConverterType.java`.

Shouldn't we have a kind of reflection config or doc accordingly ?

File: integration-tests/jaxb/src/main/resources/application.properties [hunk 1]
```diff
@@ -16,10 +16,7 @@
...
-# Quarkus
-#
-
-## include xml routes and scheme
-quarkus.native.resources.includes = person.xsd
-
+# Note: jaxb.index for the 'pojo' package is added manually since PojoPerson is a slight edge case test
+# where it does not have any JAXB annotations. Thus quarkus-jaxb cannot auto create
+# the jaxb.index for the native image
+quarkus.native.resources.includes = person.xsd,org/apache/camel/quarkus/component/jaxb/it/model/pojo/jaxb.index
```
[the remark this answers is not in the document]
Comment (jamesnetherton) [reply integration-tests/jaxb/src/main/resources/application.properties:22]: Actually it's a leftover that is not required. I'll remove it.

**Comment by jamesnetherton** [comment]
> Shouldn't we have a kind of reflection config or doc accordingly ?

Possibly not in this case because the reflective class config is being handled by `quarkus-jaxb`. Although, I am missing a test to verify this so I'll add one to make sure.

**Comment by aldettinger** [comment]
Okay great and it makes sense now that `quarkus-jaxb` would bring some reflective config too :+1:

**Comment by jamesnetherton** [comment]
Latest commit adds a new test `marshalWithObjectFactory` and includes some docs about `ObjectFactory` and reflection.

**Review by aldettinger — APPROVED** [review]
Looks great, well done @jamesnetherton :clap:
